core/src/main/java/com/scalar/db/common/TableMetadataManager.java

```diff
@@ -20,6 +21,14 @@ public class TableMetadataManager {
   private final LoadingCache<TableKey, Optional<TableMetadata>> tableMetadataCache;
 
   public TableMetadataManager(Admin admin, long cacheExpirationTimeSecs) {
+    this(
+        (key) -> Optional.ofNullable(admin.getTableMetadata(key.namespace, key.table)),
+        cacheExpirationTimeSecs);
+  }
+
+  public TableMetadataManager(
+      ThrowableFunction<Optional<TableMetadata>, TableKey, Exception> getTableMetadataFunc,
```

Comment:
After this change, we can also specify a function to retrieve a table metadata in `TableMetadataManager`.